testpar/t_filters_parallel.c

```diff
@@ -592,15 +602,14 @@ verify_chunk_opt_status(size_t num_dsets, test_mode_t test_mode, bool any_io, bo
                          * should be scalar I/O for allocation in addition to vector I/O for the actual data.
                          * If we're reading from an unallocated dataset then there should be no actual I/O.
                          * Otherwise there should only be vector I/O. */
-                        if (did_alloc)
-                            VRFY((H5D_SCALAR_IO | H5D_VECTOR_IO) == actual_sel_io_mode_reduced,
-                                 "verified actual selection I/O mode was scalar and vector I/O");
-                        else if (unalloc_read)
+                        /* Changes: for allocation, the reported I/O is vector I/O */
```

Comment:
Same here. Explain that both collective filling/allocation and filtered I/O use vector I/O so only that is reported whether or not allocation happened.